software/base/src/main/java/org/apache/brooklyn/entity/software/base/ChildSoftwareProcess.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.entity.software.base;
+
+import com.google.common.annotations.Beta;
+
+import org.apache.brooklyn.api.entity.ImplementedBy;
+import org.apache.brooklyn.location.ssh.SshMachineLocation;
+
+/**
+ * An entity that has the same functionality as {@link VanillaSoftwareProcess}
+ * but starts in the same {@link SshMachineLocation machine} as its parent.
```

Comment:
I thought this would happen automatically with vanillasoftwareprocess? The parent is given a ProvisioningLocation and creates the machine; the child is passed the machine in the start() call. Maybe things changed with the way it now looks in the parent's locations (to find those passed in via an EntitySpec)?